src/main/kotlin/org/rust/lang/core/completion/LookupElements.kt

```diff
@@ -0,0 +1,101 @@
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.parentOfType
+import org.rust.lang.core.type.resolvedType
+
+fun RustNamedElement.createLookupElement(): LookupElement {
+    return when (this) {
+        is RustFnItem            -> createLookupElement()
+        is RustImplMethodMember  -> createLookupElement()
+        is RustTraitMethodMember -> createLookupElement()
+        is RustConstItem         -> createLookupElement()
+        is RustStaticItem        -> createLookupElement()
+        is RustPath              -> createLookupElement()
+        is RustUseGlob           -> createLookupElement()
+        is RustTraitItem         -> createLookupElement()
+        is RustStructItem        -> createLookupElement()
+        is RustEnumVariant       -> createLookupElement()
+        is RustFieldDecl         -> createLookupElement()
+        else                     -> LookupElementBuilder.createWithIcon(this).withLookupString(name ?: "")
+    }
+}
+
+fun RustFnItem.createLookupElement(): LookupElement {
```

Comment:
This functions should be private. Alternatively, they all can be just inlined in that `when`. Overloading of extension methods can seem a bit confusing.